webui/react/src/hooks/useMetricNames.ts

```diff
@@ -6,10 +6,17 @@ import { detApi } from 'services/apiConfig';
 import { readStream } from 'services/utils';
 import { Metric, MetricType } from 'types';
 import { isEqual } from 'utils/data';
+import { Loadable, Loaded, NotLoaded } from 'utils/loadable';
 import { alphaNumericSorter } from 'utils/sort';
 
-const useMetricNames = (experimentIds: number[], errorHandler?: (e: unknown) => void): Metric[] => {
-  const [metrics, setMetrics] = useState<Metric[]>([]);
+const useMetricNames = (
+  experimentIds: number[],
+  errorHandler?: (e: unknown) => void,
+): {
+  isLoaded: boolean;
+  metrics: Metric[];
+} => {
```

Comment:
I think returning a Loadable object here instead of splitting it into a boolean and a value would look cleaner

Reply:
Agreed and updated.